File: backend/src/Shalom.Application/Contracts/TodayFastingDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// Fasting pillar of the Today aggregate: the open session (if any), today's
/// eating window, whether "now" sits inside it (local, ADR-004), and the
/// schedule's target for today.
/// </summary>
public record TodayFastingDto(
    FastingSessionDto? Current,
    TodayWindowDto TodayWindow,
    bool WindowOpen,
    int TargetHours);

File: frontend/projects/api/src/lib/models/today-fasting-dto.ts
import { FastingSessionDto } from './fasting-session-dto';
import { TodayWindowDto } from './today-window-dto';

/**
 * Fasting pillar of the Today aggregate: the open session (if any), today's
 * eating window, whether "now" sits inside it (computed server-side in the
 * schedule's timezone, ADR-004), and today's target hours.
 */
export interface TodayFastingDto {
  readonly current: FastingSessionDto | null;
  readonly todayWindow: TodayWindowDto;
  readonly windowOpen: boolean;
  readonly targetHours: number;
}
